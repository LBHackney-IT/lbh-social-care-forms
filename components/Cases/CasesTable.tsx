import forms from 'data/flexibleForms';
import { Case } from 'types';
import { formatDate, isDateValid } from 'utils/date';

import CaseLink from './CaseLink';

const CaseNotePersonId = ({ recordId, personId }: Case) => (
  <td key={`${recordId}-personId`} className="govuk-table__cell">
    {personId}
  </td>
);

const CaseNoteDate = ({ recordId, dateOfEvent, caseFormTimestamp }: Case) => (
  <td
    key={`${recordId}-date`}
    className="govuk-table__cell govuk--timestamp"
    style={{ width: '120px' }}
  >
    {(dateOfEvent && formatDate(dateOfEvent)) ||
      (caseFormTimestamp && formatDate(caseFormTimestamp))}
  </td>
);

const CaseNoteTitle = ({ recordId, formName, caseFormData }: Case) => {
  // 1. handle case notes
  if (
    ['ASC_case_note', 'CFS_case_note'].includes(
      caseFormData?.form_name_overall
    ) &&
    'Case Note - '
  )
    return (
      <td key={`${recordId}-title`} className="govuk-table__cell">
        Case note: {formName}
        <br />
        <p className="lbh-body-s govuk-!-margin-top-2">
          {caseFormData?.case_note_title}
        </p>
      </td>
    );

  // 2. handle flexible form submissions
  const form = forms.find((form) => form.id === formName);
  if (form)
    return (
      <td key={`${recordId}-title`} className="govuk-table__cell">
        {form.name}
      </td>
    );

  // 3. handle everything else
  return (
    <td key={`${recordId}-title`} className="govuk-table__cell">
      {formName}
    </td>
  );
};

const CaseNoteOfficer = ({ recordId, officerEmail }: Case) => (
  <td key={`${recordId}-officer`} className="govuk-table__cell">
    {officerEmail}
  </td>
);

const CaseNoteAction = ({
  recordId,
  caseFormUrl,
  caseFormData,
  formName,
  personId,
  formType,
  title,
}: Case) => (
  <td
    key={`${recordId}-action`}
    className="govuk-table__cell govuk-table__cell--numeric"
    style={{ width: '50px', textAlign: 'center' }}
  >
    <CaseLink
      recordId={recordId}
      externalUrl={caseFormUrl}
      caseFormData={caseFormData}
      formName={formName}
      personId={personId}
      formType={formType}
      title={title}
    />
  </td>
);

const CaseNoteName = ({ recordId, firstName, lastName }: Case) => (
  <td key={`${recordId}-name`} className="govuk-table__cell">
    {firstName} {lastName}
  </td>
);

const CaseNoteBirthday = ({ recordId, dateOfBirth }: Case) => (
  <td key={`${recordId}-birthday`} className="govuk-table__cell">
    {dateOfBirth && isDateValid(dateOfBirth) && dateOfBirth}
  </td>
);

const tableEntities = {
  person_id: { text: 'Person ID', component: CaseNotePersonId },
  first_name: { text: 'Client name', component: CaseNoteName },
  date_of_birth: { text: 'Date of birth', component: CaseNoteBirthday },
  date_of_event: { text: 'Date created', component: CaseNoteDate },
  formName: { text: 'Record type', component: CaseNoteTitle },
  officer_email: { text: 'Created by', component: CaseNoteOfficer },
  action: { text: 'Action', component: CaseNoteAction },
};

export type CaseTableColumns = keyof typeof tableEntities;

interface Props {
  records: Case[];
  columns: CaseTableColumns[];
}

const CasesTable = ({ records, columns }: Props): React.ReactElement => (
  <table className="govuk-table">
    <thead className="govuk-table__head">
      <tr className="govuk-table__row">
        {columns.map((column) => (
          <th key={column} scope="col" className="govuk-table__header">
            {tableEntities[column].text}{' '}
          </th>
        ))}
      </tr>
    </thead>
    <tbody className="govuk-table__body">
      {records.map((record) => (
        <tr className="govuk-table__row" key={record.recordId}>
          {columns.map((column) => tableEntities[column].component(record))}
        </tr>
      ))}
    </tbody>
  </table>
);

export default CasesTable;
