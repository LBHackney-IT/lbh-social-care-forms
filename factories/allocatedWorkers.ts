import { Factory } from 'fishery';

import { Allocation } from 'types';

export const allocationFactory = Factory.define<Allocation>(({ sequence }) => ({
  id: sequence,
  allocatedWorkerTeam: 'Safeguarding and Reviewing Team',
  allocatedWorker: 'Officer Name',
  allocationEndDate: '2019-03-28 00:00:00',
  allocationStartDate: '2019-03-28 00:00:00',
  workerType: 'Consultant Social Worker',
  personId: 1234,
  caseStatus: 'Open',
  personName: 'foo',
  personAddress: 'the address',
  personDateOfBirth: '2020-03-20',
  ragRating: 'amber',
}));

export const mockedAllocation = allocationFactory.build();
export const mockedAllocations = [allocationFactory.build()];
