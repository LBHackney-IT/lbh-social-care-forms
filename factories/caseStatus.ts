import { Factory } from 'fishery';
import {
  CaseStatus,
  CaseStatusFields,
  AddCaseStatusFormData,
  CaseStatusAnswerDisplay,
} from 'types';

export const mockedCaseStatusFactory = Factory.define<CaseStatus>(
  ({ sequence }) => ({
    id: sequence,
    type: 'CIN',
    answers: [mockedStatusField.build()],
    startDate: '2021-01-01T02:00:00Z',
    endDate: '2021-12-01T02:00:00Z',
    notes: 'a lot to note',
  })
);

export const mockedCaseStatusAddRequest = Factory.define<AddCaseStatusFormData>(
  ({ sequence }) => ({
    personId: sequence,
    type: 'CIN',
    fields: [{ name: 'test', selected: 'test' }],
    startDate: '2021-01-01T02:00:00Z',
    endDate: '2021-12-01T02:00:00Z',
    notes: 'a lot to note',
    createdby: 'test@testington.com',
  })
);

export const mockedStatusField = Factory.define<CaseStatusFields>(() => ({
  option: 'placementType',
  value: 'K1',
  startDate: '2020-08-01',
  createdAt: '2021-10-11T10:54:32Z',
}));

export const mockedCaseStatusAnswers = Factory.define<CaseStatusAnswerDisplay>(
  () => ({
    startDate: '2021-09-10',
    endDate: '2021-10-10',
    status: [mockedStatusField.build()],
  })
);
