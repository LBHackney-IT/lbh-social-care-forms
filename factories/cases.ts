import { Factory } from 'fishery';

import { Case } from 'types';

export const caseFactory = Factory.define<Case>(({ sequence }) => ({
  recordId: sequence.toString(),
  personId: 123,
  dateOfEvent: '25/10/2020 13:49:43',
  formName: 'foorm',
  officerEmail: 'Fname.Lname@hackney.gov.uk',
  caseFormData: {
    note: 'I am the note',
    first_name: 'Foo',
    last_name: 'Bar',
    timestamp: '11/03/2021 16:51:56',
    mosaic_id: 123,
    form_name_overall: 'foo',
    worker_email: 'worker@hackney.gov.uk',
    form_name: 'Foo bar',
    context_flag: 'A',
    date_of_event: '25/10/2020 13:49:43',
  },
  deleted: false,
  isImported: false,
}));

export const mockedNote = caseFactory.build({ caseFormUrl: 'https://foo.bar' });
export const mockedAllocationNote = caseFactory.build({
  formName: 'Worker allocated',
  caseFormData: {
    allocation_id: 321,
    created_by: 'foo@hackney.gov.uk',
    form_name_overall: 'API_Allocation',
    form_name: 'Worker allocated',
  },
});
export const mockedDeallocationNote = caseFactory.build({
  formName: 'Worker allocated',
  caseFormData: {
    allocation_id: 321,
    created_by: 'foo@hackney.gov.uk',
    deallocation_reason: 'deallocated because',
    form_name_overall: 'API_Deallocation',
    form_name: 'Worker deallocated',
  },
});
export const mockedCaseNote = caseFactory.build({
  caseFormData: {
    form_name_overall: 'ASC_case_note',
    case_note_title: 'i am a case title',
  },
});
export const mockedDeletedCaseNote = caseFactory.build({
  caseFormData: {
    form_name_overall: 'ASC_case_note',
    case_note_title: 'i am a case title',
  },
  formName: 'adult-case-note',
  deleted: true,
});
export const mockedWarningNoteCase = caseFactory.build({
  formName: 'Warning Note',
  caseFormData: {
    form_name_overall: 'API_WarningNote',
    warning_note_id: 456,
  },
});
