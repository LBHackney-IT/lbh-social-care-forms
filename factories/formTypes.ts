import { Factory } from 'fishery';
import { FormValue, FormStatusValue } from 'types';

export const mockedFormValueFactory = Factory.define<FormValue>(() => ({
  name: 'test',
  description: 'description',
  options: [mockedFormStatusValueFactory.build()],
}));

export const mockedFormStatusValueFactory = Factory.define<FormStatusValue>(
  () => ({
    name: 'N1',
    description: 'Noo 1',
  })
);

export const mockedFormValue = [mockedFormValueFactory.build()];
