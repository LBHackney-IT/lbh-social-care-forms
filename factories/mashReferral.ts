import { Factory } from 'fishery';
import { MashReferral, ReferralStage } from 'types';
import { mockedWorker } from './workers';

export const mashReferralFactory = Factory.define<MashReferral>(
  ({ sequence }) => ({
    id: sequence.toString(),
    referrer: 'hardcoded-referrer',
    requestedSupport: 'test-requested-support',
    assignedTo: mockedWorker,
    createdAt: '2021-10-27T09:32:17.319Z',
    clients: ['test-client'],
    referralDocumentURI: 'test-referral-document-URI',
    stage: ReferralStage.INITIAL,
    contactUrgentContactRequired: false,
    contactCreatedAt: '2021-10-28T10:55:45.097Z',
    initialReferralCategory: 'test-referral-category',
    initialDecision: 'test-initial-decision',
    initialUrgentContactRequired: false,
    initialCreatedAt: '2021-11-01T15:12:05.097Z',
    screeningDecision: 'test-screening-decision',
    screeningUrgentContactRequired: false,
    screeningCreatedAt: '2021-11-03T18:52:05.097Z',
    finalDecision: 'test-final-decision',
    finalReferralCategory: 'test-final-referral-category',
    finalUrgentContactRequired: false,
    finalCreatedAt: '2021-11-18T15:10:05.097Z',
  })
);

export const mockedMashReferral = mashReferralFactory.build();
export const mockedMashReferrals = [mashReferralFactory.build()];
