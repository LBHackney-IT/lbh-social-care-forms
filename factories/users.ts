import { Factory } from 'fishery';

import { User } from 'types';

export const userFactory = Factory.define<User>(() => ({
  name: 'foo',
  hasAdminPermissions: true,
  hasChildrenPermissions: true,
  hasAdultPermissions: true,
  hasDevPermissions: false,
  isAuditable: false,
  email: 'foo@bar.com',
  permissionFlag: 'A',
  isAuthorised: true,
  isInWorkflowsPilot: false,
  isInSafeguardingReviewing: false,
  isInPlacementManagementUnit: false,
}));

export const mockedOnlyAdultUser = userFactory.build({
  hasAdminPermissions: false,
  hasChildrenPermissions: false,
  hasDevPermissions: false,
});

export const mockedOnlyChildUser = userFactory.build({
  hasAdminPermissions: false,
  hasAdultPermissions: false,
  hasDevPermissions: false,
});

export const mockedAdminUser = userFactory.build({
  hasDevPermissions: false,
});

export const mockedUser = userFactory.build();

export const mockedUserInWorkflowsPilot = userFactory.build({
  isInWorkflowsPilot: true,
});
