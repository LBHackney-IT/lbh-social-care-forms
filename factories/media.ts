import { Media } from 'types';
import { mockedResident } from 'factories/residents';
import { mockedWorker } from 'factories/workers';
import { mockSubmission } from './submissions';

export const mockedMedia: Media = {
  mediaId: 'abc123',
  title: 'Example media file',
  resident: mockedResident,
  uploadedBy: mockedWorker,
  uploadedAt: '2021-06-21T12:00:00.000Z',
  submission: mockSubmission,
  files: {
    original: {
      type: 'image/png',
      size: 20000000,
      url: 'http://placehold.it/3000x2000',
    },
    medium: {
      type: 'image/png',
      size: 200000,
      url: 'http://placehold.it/500x333',
    },
    thumbnail: {
      type: 'image/png',
      size: 20000,
      url: 'http://placehold.it/100x150',
    },
  },
};
