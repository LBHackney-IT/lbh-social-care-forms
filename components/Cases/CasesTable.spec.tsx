import { render } from '@testing-library/react';

import CasesTable from './CasesTable';
import {
  mockedNote,
  mockedCaseNote,
  mockedAllocationNote,
  mockedDeletedCaseNote,
} from 'factories/cases';
import { mockedUser } from 'factories/users';

describe('CasesTable component', () => {
  it('should render properly', () => {
    const props = {
      records: [mockedNote, mockedAllocationNote, mockedCaseNote],
      user: mockedUser,
    };

    const { asFragment } = render(
      <CasesTable
        {...props}
        columns={['date_of_event', 'formName', 'officer_email', 'action']}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('should not display deleted case notes for non admin users', () => {
    mockedUser.hasAdminPermissions = false;
    mockedUser.hasDevPermissions = false;
    const props = {
      records: [
        mockedNote,
        mockedAllocationNote,
        mockedCaseNote,
        mockedDeletedCaseNote,
      ],
      user: mockedUser,
    };

    const { asFragment } = render(
      <CasesTable
        {...props}
        columns={['date_of_event', 'formName', 'officer_email', 'action']}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('should display deleted case notes for admin users', () => {
    mockedUser.hasAdminPermissions = true;
    mockedUser.hasDevPermissions = true;
    const props = {
      records: [
        mockedNote,
        mockedAllocationNote,
        mockedCaseNote,
        mockedDeletedCaseNote,
      ],
      user: mockedUser,
    };

    const { asFragment, getByText } = render(
      <CasesTable
        {...props}
        columns={['date_of_event', 'formName', 'officer_email', 'action']}
      />
    );
    expect(getByText(/(deleted)/)).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
