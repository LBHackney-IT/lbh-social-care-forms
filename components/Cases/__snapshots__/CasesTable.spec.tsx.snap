// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CasesTable component should display deleted case notes for admin users 1`] = `
<DocumentFragment>
  <table
    class="govuk-table"
  >
    <thead
      class="govuk-table__head"
    >
      <tr
        class="govuk-table__row"
      >
        <th
          class="govuk-table__header"
          scope="col"
        >
          Date created 
        </th>
        <th
          class="govuk-table__header"
          scope="col"
        >
          Record type 
        </th>
        <th
          class="govuk-table__header"
          scope="col"
        >
          Created by 
        </th>
        <th
          class="govuk-table__header"
          scope="col"
        >
          Action 
        </th>
      </tr>
    </thead>
    <tbody
      class="govuk-table__body"
    >
      <tr
        class="govuk-table__row"
      >
        <td
          class="govuk-table__cell govuk--timestamp"
          style="width: 120px;"
        >
          25 Oct 2020
        </td>
        <td
          class="govuk-table__cell"
        >
          foorm 
        </td>
        <td
          class="govuk-table__cell"
        >
          Fname.Lname@hackney.gov.uk
        </td>
        <td
          class="govuk-table__cell govuk-table__cell--numeric"
          style="width: 50px; text-align: center;"
        >
          <a
            class="lbh-link"
            href="https://foo.bar"
            rel="noreferrer noopener"
            target="_blank"
          >
            View
          </a>
        </td>
      </tr>
      <tr
        class="govuk-table__row"
      >
        <td
          class="govuk-table__cell govuk--timestamp"
          style="width: 120px;"
        >
          25 Oct 2020
        </td>
        <td
          class="govuk-table__cell"
        >
          Worker allocated 
        </td>
        <td
          class="govuk-table__cell"
        >
          Fname.Lname@hackney.gov.uk
        </td>
        <td
          class="govuk-table__cell govuk-table__cell--numeric"
          style="width: 50px; text-align: center;"
        >
          <a
            class="lbh-link"
            href="/people/123/allocations/321?recordId=2"
          >
            View
          </a>
        </td>
      </tr>
      <tr
        class="govuk-table__row"
      >
        <td
          class="govuk-table__cell govuk--timestamp"
          style="width: 120px;"
        >
          25 Oct 2020
        </td>
        <td
          class="govuk-table__cell"
        >
          Case note: foorm 
          <br />
          <p
            class="lbh-body-s govuk-!-margin-top-2"
          >
            i am a case title
          </p>
        </td>
        <td
          class="govuk-table__cell"
        >
          Fname.Lname@hackney.gov.uk
        </td>
        <td
          class="govuk-table__cell govuk-table__cell--numeric"
          style="width: 50px; text-align: center;"
        >
          <a
            class="lbh-link"
            href="/people/123/records/4"
          >
            View
          </a>
        </td>
      </tr>
      <tr
        class="govuk-table__row"
      >
        <td
          class="govuk-table__cell govuk--timestamp"
          style="width: 120px;"
        >
          25 Oct 2020
        </td>
        <td
          class="govuk-table__cell"
        >
          Case note: adult-case-note (deleted)
          <br />
          <p
            class="lbh-body-s govuk-!-margin-top-2"
          >
            i am a case title
          </p>
        </td>
        <td
          class="govuk-table__cell"
        >
          Fname.Lname@hackney.gov.uk
        </td>
        <td
          class="govuk-table__cell govuk-table__cell--numeric"
          style="width: 50px; text-align: center;"
        >
          <a
            class="lbh-link"
            href="/people/123/records/5"
          >
            View
          </a>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`CasesTable component should not display deleted case notes for non admin users 1`] = `
<DocumentFragment>
  <table
    class="govuk-table"
  >
    <thead
      class="govuk-table__head"
    >
      <tr
        class="govuk-table__row"
      >
        <th
          class="govuk-table__header"
          scope="col"
        >
          Date created 
        </th>
        <th
          class="govuk-table__header"
          scope="col"
        >
          Record type 
        </th>
        <th
          class="govuk-table__header"
          scope="col"
        >
          Created by 
        </th>
        <th
          class="govuk-table__header"
          scope="col"
        >
          Action 
        </th>
      </tr>
    </thead>
    <tbody
      class="govuk-table__body"
    >
      <tr
        class="govuk-table__row"
      >
        <td
          class="govuk-table__cell govuk--timestamp"
          style="width: 120px;"
        >
          25 Oct 2020
        </td>
        <td
          class="govuk-table__cell"
        >
          foorm 
        </td>
        <td
          class="govuk-table__cell"
        >
          Fname.Lname@hackney.gov.uk
        </td>
        <td
          class="govuk-table__cell govuk-table__cell--numeric"
          style="width: 50px; text-align: center;"
        >
          <a
            class="lbh-link"
            href="https://foo.bar"
            rel="noreferrer noopener"
            target="_blank"
          >
            View
          </a>
        </td>
      </tr>
      <tr
        class="govuk-table__row"
      >
        <td
          class="govuk-table__cell govuk--timestamp"
          style="width: 120px;"
        >
          25 Oct 2020
        </td>
        <td
          class="govuk-table__cell"
        >
          Worker allocated 
        </td>
        <td
          class="govuk-table__cell"
        >
          Fname.Lname@hackney.gov.uk
        </td>
        <td
          class="govuk-table__cell govuk-table__cell--numeric"
          style="width: 50px; text-align: center;"
        >
          <a
            class="lbh-link"
            href="/people/123/allocations/321?recordId=2"
          >
            View
          </a>
        </td>
      </tr>
      <tr
        class="govuk-table__row"
      >
        <td
          class="govuk-table__cell govuk--timestamp"
          style="width: 120px;"
        >
          25 Oct 2020
        </td>
        <td
          class="govuk-table__cell"
        >
          Case note: foorm 
          <br />
          <p
            class="lbh-body-s govuk-!-margin-top-2"
          >
            i am a case title
          </p>
        </td>
        <td
          class="govuk-table__cell"
        >
          Fname.Lname@hackney.gov.uk
        </td>
        <td
          class="govuk-table__cell govuk-table__cell--numeric"
          style="width: 50px; text-align: center;"
        >
          <a
            class="lbh-link"
            href="/people/123/records/4"
          >
            View
          </a>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`CasesTable component should render properly 1`] = `
<DocumentFragment>
  <table
    class="govuk-table"
  >
    <thead
      class="govuk-table__head"
    >
      <tr
        class="govuk-table__row"
      >
        <th
          class="govuk-table__header"
          scope="col"
        >
          Date created 
        </th>
        <th
          class="govuk-table__header"
          scope="col"
        >
          Record type 
        </th>
        <th
          class="govuk-table__header"
          scope="col"
        >
          Created by 
        </th>
        <th
          class="govuk-table__header"
          scope="col"
        >
          Action 
        </th>
      </tr>
    </thead>
    <tbody
      class="govuk-table__body"
    >
      <tr
        class="govuk-table__row"
      >
        <td
          class="govuk-table__cell govuk--timestamp"
          style="width: 120px;"
        >
          25 Oct 2020
        </td>
        <td
          class="govuk-table__cell"
        >
          foorm 
        </td>
        <td
          class="govuk-table__cell"
        >
          Fname.Lname@hackney.gov.uk
        </td>
        <td
          class="govuk-table__cell govuk-table__cell--numeric"
          style="width: 50px; text-align: center;"
        >
          <a
            class="lbh-link"
            href="https://foo.bar"
            rel="noreferrer noopener"
            target="_blank"
          >
            View
          </a>
        </td>
      </tr>
      <tr
        class="govuk-table__row"
      >
        <td
          class="govuk-table__cell govuk--timestamp"
          style="width: 120px;"
        >
          25 Oct 2020
        </td>
        <td
          class="govuk-table__cell"
        >
          Worker allocated 
        </td>
        <td
          class="govuk-table__cell"
        >
          Fname.Lname@hackney.gov.uk
        </td>
        <td
          class="govuk-table__cell govuk-table__cell--numeric"
          style="width: 50px; text-align: center;"
        >
          <a
            class="lbh-link"
            href="/people/123/allocations/321?recordId=2"
          >
            View
          </a>
        </td>
      </tr>
      <tr
        class="govuk-table__row"
      >
        <td
          class="govuk-table__cell govuk--timestamp"
          style="width: 120px;"
        >
          25 Oct 2020
        </td>
        <td
          class="govuk-table__cell"
        >
          Case note: foorm 
          <br />
          <p
            class="lbh-body-s govuk-!-margin-top-2"
          >
            i am a case title
          </p>
        </td>
        <td
          class="govuk-table__cell"
        >
          Fname.Lname@hackney.gov.uk
        </td>
        <td
          class="govuk-table__cell govuk-table__cell--numeric"
          style="width: 50px; text-align: center;"
        >
          <a
            class="lbh-link"
            href="/people/123/records/4"
          >
            View
          </a>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
